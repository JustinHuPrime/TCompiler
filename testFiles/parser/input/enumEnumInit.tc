module foo;

enum bar {
  A,
  B,
  C,
};

enum baz {
  D = bar::C,
  E,
  F,
};