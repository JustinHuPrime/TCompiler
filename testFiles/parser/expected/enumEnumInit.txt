testFiles/parser/input/enumEnumInit.tc (code):
FILE(1, 1, STAB(ENTRY(bar, ENUM(testFiles/parser/input/enumEnumInit.tc, 3, 1, CONSTANT(A, 0), CONSTANT(B, 1), CONSTANT(C, 2))), ENTRY(baz, ENUM(testFiles/parser/input/enumEnumInit.tc, 9, 1, CONSTANT(D, 2), CONSTANT(E, 3), CONSTANT(F, 4)))), MODULE(1, 1, ID(1, 8, foo, REFERENCES())), ENUMDECL(3, 1, ID(3, 6, bar, REFERENCES(testFiles/parser/input/enumEnumInit.tc, 3, 1)), ID(4, 3, A, REFERENCES(testFiles/parser/input/enumEnumInit.tc, 4, 3)), ID(5, 3, B, REFERENCES(testFiles/parser/input/enumEnumInit.tc, 5, 3)), ID(6, 3, C, REFERENCES(testFiles/parser/input/enumEnumInit.tc, 6, 3)), (null), (null), (null)), ENUMDECL(9, 1, ID(9, 6, baz, REFERENCES(testFiles/parser/input/enumEnumInit.tc, 9, 1)), ID(10, 3, D, REFERENCES(testFiles/parser/input/enumEnumInit.tc, 10, 3)), ID(11, 3, E, REFERENCES(testFiles/parser/input/enumEnumInit.tc, 11, 3)), ID(12, 3, F, REFERENCES(testFiles/parser/input/enumEnumInit.tc, 12, 3)), SCOPEDID(10, 7, bar::C, REFERENCES()), (null), (null)))
